application/mounts.py

```diff
@@ -42,11 +42,21 @@ def _add_mount(self, monitor, mount):
 		mount_uri = mount.get_root().get_uri()
 		volume = mount.get_volume()
 
+		# check if mount has been added as volume already
+		mount_already_in_volume_list = False
```

Comment:
I verified the weird behaviour ("mount-added" callback getting called 
twice) is the same in Debian Jessie as in Ubuntu 14.04 and my code 
handles this as far as I can tell.

Interestingly, I found this bug: 
https://bugzilla.gnome.org/show_bug.cgi?id=730347 which sounds quite 
similar to what I see for MTP and Gphoto2 GIO/GVFS mounts.

Christian

On 15/05/15 21:56, Mladen Mijatov wrote:
>
> In application/mounts.py 
> <https://github.com/MeanEYE/Sunflower/pull/112#discussion_r30445418>:
>
> > @@ -42,11 +42,21 @@ def _add_mount(self, monitor, mount):
> >  		mount_uri = mount.get_root().get_uri()
> >  		volume = mount.get_volume()
> >
> > +		# check if mount has been added as volume already
> > +		mount_already_in_volume_list = False
>
> I think this was a bug in GIO as 14.04 is quite old. I remember fixing 
> the same issue with duplicate mounts as I thought GIO was suppose to 
> work this way. After they fixed this issue in GIO they created a 
> problem for us. If possible, please test this on some newer version 
> just to make sure we are not creating another problem.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/MeanEYE/Sunflower/pull/112/files#r30445418>.
>